File: 1.0/src/c/cz/mg/vulkan/PFN_vkAcquireNextImage2KHX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkAcquireNextImage2KHX_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pAcquireInfo, jlong pImageIndex, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkAcquireNextImage2KHX f = (PFN_vkAcquireNextImage2KHX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkAcquireNextImageInfoKHX*)jniLongToPointer(pAcquireInfo)),
        ((uint32_t*)jniLongToPointer(pImageIndex))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkAcquireNextImage2KHX.java
package cz.mg.vulkan;

public class PFNvkAcquireNextImage2KHX extends VkFunctionPointer {
    public PFNvkAcquireNextImage2KHX() {
    }

    protected PFNvkAcquireNextImage2KHX(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkAcquireNextImage2KHX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkAcquireNextImage2KHX(long value) {
        setValue(value);
    }

    public PFNvkAcquireNextImage2KHX(VkInstance instance) {
        super(instance, new VkString("vkAcquireNextImage2KHX"));
    }

    public void call(VkDevice device, VkAcquireNextImageInfoKHX pAcquireInfo, VkUInt32 pImageIndex, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pAcquireInfo != null ? pAcquireInfo.getVkAddress() : VkPointer.NULL, pImageIndex != null ? pImageIndex.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long device, long pAcquireInfo, long pImageIndex, long rval);
}
